flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -362,11 +362,22 @@ public void cancelSessionJob(
             FlinkSessionJob sessionJob, UpgradeMode upgradeMode, Configuration conf)
             throws Exception {
 
-        var jobStatus = sessionJob.getStatus().getJobStatus();
+        var sessionJobStatus = sessionJob.getStatus();
+        var jobStatus = sessionJobStatus.getJobStatus();
         var jobIdString = jobStatus.getJobId();
         Preconditions.checkNotNull(jobIdString, "The job to be suspend should not be null");
         var jobId = JobID.fromHexString(jobIdString);
         Optional<String> savepointOpt = Optional.empty();
+
+        if (ReconciliationUtils.isJobInTerminalState(sessionJobStatus)) {
+            LOG.info("Job is already in terminal state. JobID {}", jobId.toHexString());
+            return;
+        } else if (!ReconciliationUtils.isJobRunning(sessionJobStatus)) {
```

Comment:
I think we don’t need this branch here, this would prevent us from suspending failing jobs

Reply:
I have re-written the logic and added test cases accordingly. Please check.

Reply:
@gaborgsomogyi - Please kindly check.

Reply:
@gyfora - The build pipeline is failing with the below error. At this place: https://github.com/apache/flink-kubernetes-operator/actions/runs/3574448062/jobs/6009735670. 

Could you please help me here? Not sure if re-triggering solves it. 

```
flinkdeployment.flink.apache.org "session-cluster-1" deleted
flinksessionjob.flink.apache.org "flink-example-statemachine" deleted
persistentvolumeclaim "session-cluster-1-pvc" deleted
ingressclass.networking.k8s.io "nginx" deleted
No resources found
Error: Process completed with exit code 1.
```

Reply:
above issue is resolved now.